Refactor

ellezv opened the pull request:
Working from David's feedback:
- tests were refactored and now pass
- some functions were broken down in small functions for better testing
- new tests for the new functions
- exception handling updated.
- new README

chamberi committed:
initial base test to see if we produce two words

ellezv committed:
simple function to return the first two words as a string

chamberi committed:
added 2nd test to assert list of text gets created

ellezv committed:
added import io

ellezv committed:
added sample txt

chamberi committed:
wrote test to parse words into list with single words

ellezv committed:
test written for our our dictionary

chamberi committed:
built enumerator that builds out the 2-word keys and value pairs

ellezv committed:
refactor pulltext function using regex

chamberi committed:
added test to see if output file is the same as the function produced

chamberi committed:
fixed errors, added test of creeat output

chamberi committed:
Merge branch 'trigrams' of https://github.com/ellezv/trigrams into trigrams

ellezv committed:
create output works with non randomized values

chamberi committed:
added random choice of list index

ellezv committed:
create_output function updated
now takes two arguments:
file_path so we can choose a book to implement our function on
user_int to determine the maximum length of the output

ellezv committed:
added if __name__ == '__main__' block that allows us to run the script from the command line using the arguments passed after the module name

chamberi committed:
added random choice to the initial key starting position

ellezv committed:
fixed random choice for random key

ellezv committed:
refactoring:
test were refactored to accept file path.
create output was broken down in other function to allow for better testing.
exception handling for keyerror

ellezv committed:
README done

Bl41r commented on src/trigrams.py:
```diff
@@ -1 +1,57 @@
+def generate_random_key(tridict):
+    """Generate a random key."""
+    random_key = random.choice(generate_key_list(tridict))
+    create_output
```
I don't think you meant to have this line here

Bl41r commented on src/trigrams.py:
```diff
@@ -1 +1,57 @@
+            output.append(random.choice(tridict[new_key]))
+        except KeyError:
+            new_key = generate_random_key(tridict)
+            output.append(random.choice(tridict[new_key]))
```
This line is causing your major bug.  See if you can figure out why!

Bl41r commented on src/trigrams.py:
```diff
@@ -1 +1,57 @@
+def pulltext(file_path):
+    """Open a text file and turn it into a list of single word strings."""
+    text_file = io.open(file_path).read()
+    return re.findall('\w+', text_file)
```
Nice job diving into regex, I know you have seen regex before from your 301 project!  Consider not stripping out punctuation so that the text is realistic.

ellezv committed:
refactored:
regex now include punctuation for more human like output,
while loop for better control of output length,
new key bug fix

ellezv committed:
tests refactored
